hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/GenericRecordQueryReader.java

```diff
@@ -250,6 +258,10 @@ private Object readLeaf(InternalGenericRecord record, String path) {
             return null;
         }
         FieldKind kind = record.getFieldKind(path);
+        if (kind == FieldKind.COMPACT && useLazyDeserialization) {
```

Comment:
What about portable?

Reply:
If we agree on the compact case, we can easily add portable as well. Will I need to add new tests for portable?  Is there a test like CompactStreamSerializerValueReaderQuickTest for portable that will test we don't break anything?

Reply:
I think that test was copied from the Portable counter-part, see `DefaultPortableReaderQuickTest`